webapp/platform/types/src/posts.ts

```diff
@@ -87,6 +87,7 @@ export type Post = {
     pending_post_id: string;
     reply_count: number;
     file_ids?: string[];
+    file_client_ids?: string[];
```

Comment:
@hmhealey 
isPostUploadingFile is used from [comparePosts](https://github.com/sk409/mattermost-server/blob/MM-37606_Queue-sending-message-after-file-upload-finish/webapp/channels/src/packages/mattermost-redux/src/utils/post_utils.ts#L160).

I think comparePosts doesn't always have access to the globalState, so I think the file_client_ids are necessary.

Reply:
Oh yeah, that's a good point. That's another reason for me to regret that we have to store sorted post IDs in the store instead of calculating the order with a selector since I realized a better way I could've written that after I originally wrote that 😛

I guess that should be fine then as long as we're sorting pending posts above ones with queued file uploads. I also just remembered that we have some other non-server fields on here like `failed` and `user_activity_posts`, so one more wouldn't hurt